### PR TITLE
creating proxy canary job for 4.2

### PR DESCRIPTION
**Pull request by ewolinetz**
This is to add the aws e2e proxy test to the list of canary jobs run periodically

### CODE REVIEW

**Comment by ewolinetz** [comment]
@wking does this look correctly plumbed out? I didn't see anything specific references to the IPI job template, will we need to update this again when we move to a UPI based install with https://github.com/openshift/release/pull/4719 ?

File: ci-operator/jobs/openshift/release/openshift-release-release-4.2-periodics.yaml [hunk 1]
```diff
@@ -553,6 +553,87 @@ periodics:
...
+      name: ""
+      resources:
+        requests:
+          cpu: 10m
+      volumeMounts:
+      - mountPath: /usr/local/e2e-aws-proxy-cluster-profile
+        name: cluster-profile
+      - mountPath: /usr/local/e2e-aws-proxy
+        name: job-definition
+        subPath: cluster-launch-installer-e2e.yaml
```
Comment (wking) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.2-periodics.yaml]: This is good for this PR, but we'll need to bump it to point to UPI when we rebase #4719 on top.

**Comment by wking** [comment]
Looks good to me; we'll see how the rehearsal plays out.

**Comment by wking** [comment]
[Proxy failed in e2e][1], but it [was using a proxy][2].  Errors may be real or flakes, but the proxy bit works.  Good enough for a canary.

/lgtm

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/5098/rehearse-5098-canary-openshift-ocp-installer-e2e-aws-proxy-4.2/1
[2]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/5098/rehearse-5098-canary-openshift-ocp-installer-e2e-aws-proxy-4.2/1/artifacts/e2e-aws-proxy/must-gather/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-4aa5ef9a27c016523f47855c3a477ac491efd2dd9af61893c836bfb9fe093333/cluster-scoped-resources/config.openshift.io/proxies/cluster.yaml

**Pull request merged**